app/services/catalog/share_info.rb

```diff
@@ -15,7 +16,7 @@ def process
         ace.group_uuid
       end
 
-      group_names = fetch_group_names(uuids.uniq)
+      group_names = @user_context.group_names(uuids)
```

Comment:
@eclarizio why is the uniq missing? It's critical

Reply:
When I first saw this comment, I agreed, but now I'm not sure because when I went to change the spec, it involved doing something like this in the setup:
```
```

Is it? There should only be one ACE for a group ID, right?

Reply:
When I first saw this comment, I agreed, but now I'm not sure because when I went to change the spec, it involved doing something like this in the setup:
```
before do
  create(:access_control_entry, :has_read_and_update_permission, :group_uuid => uuid, :aceable => portfolio)
  create(:access_control_entry, :has_read_permission, :group_uuid => uuid, :aceable => portfolio)
end
```
But that's incorrect, we would only ever have the one ACE and we either add or remove permissions onto it. So I don't think there a way that `uuids` on line 19 could ever have duplicates (in the real world, at least).